src/odemis/gui/model/__init__.py

```diff
@@ -595,6 +597,46 @@ def __init__(self, main):
         # VA for autofocus procedure mode
         self.autofocus_active = BooleanVA(False)
 
+
+class LocalizationGUIData(MicroscopyGUIData):
+    """ Represent an interface used to only show the current data from the microscope.
+
+    It it used for handling CryoSECOM systems.
+
+    """
+
+    def __init__(self, main):
+        if main.role != "cryo-secom":
+            raise ValueError("The role of the microscope is not correct")
```

Comment:
The original assert was for checking there is a microscope object in main, not the role is cryo-secom. So it should be:
`if not hasattr(main, 'microscope'):`
`raise ValueError`

Reply:
Yes it was checking if there microscope exists. But the passed argument, main, is of type MainGUIData. At the init function of this class, there are several checks on the components of the microscope such as emitters and detectors. If any of these components does not exist, then an error is raised. So there is no point from checking if there is microscope object. It would make more sense to check if the role of this microscope object is cryo secom since this class is only for cryo secom (if i'm not wrong)

Reply:
I see what you mean, but you won't know if these microscope components are indeed there by checking its role. Besides it'd be redundant as checking for the role is already done on the tab itself in `get_display_priority`.

Reply:
First, to be exact, main also has a microscope. It's sometimes `None`. So it'd be:
`assert main.microscope is not None`
This is me would suggested Mahmoud to change it this way. Typically, if microscope is None, then the role is None too. So this check is normally just a little be extra strict.
@lazem , I don't agree with your argument that this is not needed because we already check for it. If that was valid, then all the checks are unnecessary! Of course, right now the code works, all the callers are well behaved. The checks are here for two other reasons:
* Remind the reader about the expectation of the code
* Detect issues in new code, early

So I think this is fine. Mahmoud could add back the original assert, just to make things even clearer.
The only thing that I thing could be improved is the error message. It should include the actual role. Something like `"Microscope role is %s, while expected 'cryo-secom'"`